pkg/workflows/create.go

```diff
@@ -182,6 +191,21 @@ func (s *CreateWorkloadClusterTask) Run(ctx context.Context, commandContext *tas
 		return nil
 	}
 
+	if commandContext.ClusterSpec.AWSIamConfig != nil {
+		logger.Info("Installing aws-iam-authenticator on workload cluster")
+		err = commandContext.ClusterManager.InstallAwsIamAuth(ctx, workloadCluster, commandContext.ClusterSpec)
+		if err != nil {
+			commandContext.SetError(err)
+			return nil
+		}
+		logger.V(1).Info("Generating aws-iam-authenticator kubeconfig")
+		err = commandContext.ClusterManager.GenerateAwsIamAuthKubeconfig(ctx, commandContext.BootstrapCluster, workloadCluster, commandContext.ClusterSpec)
+		if err != nil {
+			commandContext.SetError(err)
+			return nil
+		}
```

Comment:
If these two always go together,  why not group them in one single cluster manager method? That way you don't need to expose to the workflow what is needed to have iam authenticator
In fact, should we just make it part of `CreateWorkloadCluster`?

Reply:
Actually the reason I didnt add it to `CreateWorkloadCluster` was I suspect I would need CNI components to be installed before hand. Would have to test that out :) 
But i do agree with combining them into one method in clustermanager to simplify workflows 👍